### PR TITLE
[Doc] [Do-not-merge] add doc for Oauth2 authentication support in node.js client and update doc for custom AuthN/AuthZ

### LINKED ISSUE #14143: [Doc] Add doc of OAuth2 authentication for Node.js client
### What issue do you find in Pulsar docs?

File an issue here for the required doc effort. 

[Pulsar client node](https://github.com/apache/pulsar-client-node) version 1.6 is about to release with the node.js client support of OAuth 2.0.  See https://github.com/apache/pulsar-client-node/pull/190 for more details.

### What is your suggestion?

Add content for node.js client in the following OAuth 2.0 topic.
https://pulsar.apache.org/docs/en/security-oauth2/

### Do you have any references?

https://github.com/apache/pulsar-client-node/pull/190
https://github.com/streamnative/examples/pull/92/files
https://github.com/streamnative/doc-cloud/pull/130/files

### Would you like to fix this issue?

Sure, I'd love to.

### Note

- [X] I have researched my question.

### PR DESCRIPTION
**Pull request by momo-jun**
### Modifications
1. To align with the latest release of node.js client 1.6.2 and fix #14143, add doc for Oauth2 authentication support in node.js client.
2. To align with #14044 for 2.11, update doc for extended AuthN/AuthZ by replacing outdated info and code samples with github URLs, adding more auth providers with links to examples.
3. Fix minor writing issues in the content involved.

The preview looks good.
image
image

### Documentation

- [x] `doc`

### CODE REVIEW

**Comment by momo-jun** [comment]
Hi @tuteng @Anonymitaet can you please review this doc PR? Thank you.

File: site2/docs/security-extending.md [hunk 1]
```diff
@@ -23,173 +23,54 @@ PulsarClient client = PulsarClient.builder()
...
-     *             if the credentials are not valid
-     */
-    String authenticate(AuthenticationDataSource authData) throws AuthenticationException;
-
-}
-```
+For the implementation of the `org.apache.pulsar.broker.authentication.AuthenticationProvider` interface, refer to [here](https://github.com/apache/pulsar/blob/master/pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProvider.java).
 
-The following is the example for Broker authentication plugins:
+You can find the following examples for different Broker authentication plugins:
```
Comment (Anonymitaet) [inline site2/docs/security-extending.md:53]:
```suggestion
You can find the following examples for different broker authentication plugins:
```

File: site2/docs/security-extending.md [hunk 2]
```diff
@@ -23,173 +23,54 @@ PulsarClient client = PulsarClient.builder()
...
+ * [Basic auth](https://github.com/apache/pulsar/blob/master/pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderToken.java)
 
 ## Authorization
 
 Authorization is the operation that checks whether a particular "role" or "principal" has permission to perform a certain operation.
 
-By default, you can use the embedded authorization provider provided by Pulsar. You can also configure a different authorization provider through a plugin.
-Note that although the Authentication plugin is designed for use in both the Proxy and Broker,
-the Authorization plugin is designed only for use on the Broker however the Proxy does perform some simple Authorization checks of Roles if authorization is enabled.
+By default, you can use the embedded authorization provider provided by Pulsar. You can also configure a different authorization provider through a plugin. Note that although the Authentication plugin is designed for use in both the Proxy and Broker, the Authorization plugin is designed only for use on the Broker.
```
Comment (Anonymitaet) [inline site2/docs/security-extending.md:65]:
```suggestion
By default, you can use the embedded authorization provider provided by Pulsar. You can also configure a different authorization provider through a plugin. Note that although the Authentication plugin is designed for use in both the Proxy and Broker, the Authorization plugin is designed only for use on the Broker.
```
why capitalize Broker? it is not a proper noun. (we can update the original text if it is not accurate)

File: site2/docs/security-oauth2.md [hunk 3]
```diff
@@ -153,6 +153,47 @@ params = '''
 client = Client("pulsar://my-cluster:6650", authentication=AuthenticationOauth2(params))
 ```
 
+### Node.js client
+
+To enable OAuth2 authentication in node.js client, you need to configure OAuth2 authentication.
```
Comment (Anonymitaet) [inline site2/docs/security-oauth2.md:158]:
```suggestion
To enable OAuth2 authentication in Node.js client, you need to configure OAuth2 authentication.
```
proper noun

**Review by tuteng — COMMENTED** [review]
LGTM
Also I don't think the nodejs client supporting oauth2 authentication needs to wait for pulsar 2.11 to be released, the puslar node client was released separately

**Comment by momo-jun** [comment]
Thanks @Anonymitaet and @tuteng for your comments. I've fixed them. PTAL.

**Review by Anonymitaet — APPROVED** [review]

**Review by tuteng — APPROVED** [review]

**Pull request merged**